src/Core/src/Handlers/ViewHandlerExtensions.iOS.cs

```diff
@@ -78,7 +78,22 @@ internal static Size GetDesiredSizeFromHandler(this IViewHandler viewHandler, do
 			widthConstraint = Math.Min(widthConstraint, virtualView.MaximumWidth);
 			heightConstraint = Math.Min(heightConstraint, virtualView.MaximumHeight);
 
-			var sizeThatFits = platformView.SizeThatFits(new CoreGraphics.CGSize((float)widthConstraint, (float)heightConstraint));
+			CGSize sizeThatFits;
+
+			// Calling SizeThatFits on an ImageView always returns the image's dimensions, so we need to call the extension method
+			// This also affects ImageButtons
+			if (platformView is UIImageView imageView)
+			{
+				sizeThatFits = imageView.SizeThatFitsImage(new CGSize((float)widthConstraint, (float)heightConstraint));
```

Comment:
Not really a review, but just something that I wonder if we need to have. If we start doing special cases in random extension methods, we will never find it. We probably have this all over, so more like "if we continue".

I wonder if there is a way to ask the handler to get the desired size? Maybe like a command? So assume everything is magical:

```cs
var platformRequest = new PlatformSizeThatFitsRequest(widthConstraint, heightConstraint);
var platformResponse = viewHandler.InvokeWithResult("SizeThatFits", platformRequest);
if (platformResponse is not null)
    sizeThatFits = platformResponse;
else
    sizeThatFits = platformView.SizeThatFits(new CGSize((float)widthConstraint, (float)heightConstraint));
```

With the iOS handlers having a special mapper for `ImageHandler` and `ButtonHandler`:

```cs
ImageHandler.CommandMapper["SizeThatFits"] = (IViewHandler handler, IView view, object? args) => {
	if (args is not PlatformSizeThatFitsRequest request)
				return;
    var result = handler.PlatformView.SizeThatFitsImage(request.Constraints);
    request.SetResult(result);
}
```

The `PlatformSizeThatFitsRequest` type may be something like this:

```cs
public class PlatformSizeThatFitsRequest : RetrievePlatformValueRequest<CGSize>
{
    public PlatformSizeThatFitsRequest(nfloat width, nfloat height) {
        Constraints = new CGSize(width, height);
    }
    public CGSize Constraints { get; }
}
```

I feel like we do something like this already:

https://github.com/dotnet/maui/blob/193f8ba2c7549e8fd7440b6789caef39761054f6/src/Controls/src/Core/ViewExtensions.cs#L449-L451

Reply:
Note: this is not a recommendation... just thinking aloud to see opinions and other views.

Reply:
Happy to implement either option!

Reply:
This logic feels fundamental to how our system measures images. Do we have handler scenarios that we won't want to follow the rules defined here? These methods are outside of the handler, because they are general rules to how our system measures things. 

These methods are all internal. I feel like once we have a use case where we need to differentiate the logic then we can go down that path. 

I like the handler route that @mattleibow proposes but I feel like this entire flow should just be moved outside the handler. I think the `GetDesiredSize` call on the handler itself should just be part of the `CommandMapper`. This way, if you have a handler that reuses our mappers, but not the handler, it'll just user the `GetDesiredSize` logic. This would also allow people to replace the entire `GetDesiredSize` logic themselves. I'm not saying we should do that here but that's where I think we should eventually end up here.